src/Common/ThreadPool.h

```diff
@@ -222,6 +225,10 @@ class ThreadFromGlobalPoolImpl : boost::noncopyable
         if (!initialized())
             abort();
 
+        /// Thread cannot join itself.
+        if (state->thread_id == std::this_thread::get_id())
+            abort();
```

Comment:
But this will trigger issues like the one described in ff17de531c2bb652d88ce5cc9e5821291f5cb70c (and other places that have explicit joinable() check)

Reply:
Agree. Still:
* std::thread does not allow to join the same thread. It will result in [std::system_error](https://en.cppreference.com/w/cpp/error/system_error). pthread_join also can't do it and will return EDEADLK
* if it is possible to join the same thread - it's a bug in our code, it's better to detect in than ignore

Reply:
Let's maybe try to fix known issues. Do you know other?

Reply:
>it's a bug in our code, it's better to detect in than ignore

Completely agree. But actually abort() is better then hidden hung, so it is OK to me.

>Let's maybe try to fix known issues. Do you know other?

So far I haven't found anything more (at least those that had been fixed by me).